### ISSUE TITLE
Standard deviation for transforms.Normalize

### ISSUE THREAD

**Issue by codyaustun** [kuangliu/pytorch-cifar#8]
Hey,

How did you calculate the standard deviation values for transforms.Normalize? I am getting the same means, but different standard deviations: 
```python
import numpy as np

from torchvision import datasets
from torchvision import transforms

transform_train = transforms.Compose([
#     transforms.RandomCrop(32, padding=4),
#     transforms.RandomHorizontalFlip(),
    transforms.ToTensor()
])

trainset = datasets.CIFAR10(root='data', train=True, download=True, transform=transform_train)
train_loader = torch.utils.data.DataLoader(trainset, batch_size=50_000, shuffle=True)

train = train_loader.__iter__().next()[0]

print('Mean: {}'.format(np.mean(train.numpy(), axis=(0, 2, 3))))
# Mean: [ 0.49139765  0.48215759  0.44653141]
print('STD: {}'.format(np.std(train.numpy(), axis=(0, 2, 3))))
# STD: [ 0.24703199  0.24348481  0.26158789]
```

**Comment by kuangliu** [comment]
See `get_mean_and_std(dataset)` in `utils.py`. It's OK to use a different std for training.

**Comment by wenwei202** [comment]
Should be `[0.2470, 0.2435, 0.2616]`. See [here](https://gist.github.com/weiaicunzai/e623931921efefd4c331622c344d8151).

**Comment by lishaofeng** [comment]
I think the divide operation in get_mean_and_std(dataset) function should be 255. while not the len(dataset)

**Comment by renyiryry** [comment]
> I think the divide operation in get_mean_and_std(dataset) function should be 255. while not the len(dataset)

It should be len(dataset) because the code is summing up the std for all images. Dividing by len(dataset) gives the averaged std.

**Comment by renyiryry** [comment]
> Should be `[0.2470, 0.2435, 0.2616]`. See [here](https://gist.github.com/weiaicunzai/e623931921efefd4c331622c344d8151).

Not really. The way you computed gives the std of the tensor of size [50000, 32, 32]. However, the desired quantity is the "averaged" std across all the images. Also, when computing the std for a single image, degree of freedom should be 1. (See https://github.com/kuangliu/pytorch-cifar/blob/master/utils.py)

**Comment by ZaneLau-Info** [comment]
for inputs, targets in dataloader:
        for i in range(3):
            mean[i] += inputs[:,i,:,:].mean()
            std[i] += inputs[:,i,:,:].std()
这是因为他计算方差时，只是让像素值减去了一个batch中相应通道的像素的均值，而不是相应通道所有像素的均值。这是不严谨的。好在作者也说了“方差不影响训练结果”。
